### PR TITLE
Add NeutralResourcesLanguageAttribute to corefx libraries

### LINKED ISSUE #13507: System.Private.CoreLib assembly missing `NeutralResourcesLanguage("en-US")`
See https://github.com/dotnet/coreclr/issues/26472#issuecomment-536176848

Appears to be a regression in 3.0.  Not sure if this is intentional or not, but it does seem to be observable.

/cc @tarekgh

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes https://github.com/dotnet/runtime/issues/13507

### CODE REVIEW

**Comment by tarekgh** [comment]
@ericstj could you please have a look? Thanks.

**Comment by danmosemsft** [comment]
It would be good to get this in preview 8 (snap 15th) in case there is some subtle fallout.

**Comment by tarekgh** [comment]
> It would be good to get this in preview 8 (snap 15th) in case there is some subtle fallout.

Yes that is the goal. looks I have some minor issues with it in the CI but shouldn't be hard to fix and get it in preview 8.

File: src/coreclr/src/System.Private.CoreLib/src/System/Internal.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@
 using System;
```
Comment (akoeplinger) [inline src/coreclr/src/System.Private.CoreLib/src/System/Internal.cs:13]: given this file is identical between mono and coreclr, would you mind moving it to https://github.com/dotnet/runtime/tree/master/src/libraries/System.Private.CoreLib/src/Internal so we can share it?

maybe rename it to a more descriptive name like `AssemblyAttributes.cs` or something like that.

File: src/libraries/System.Resources.ResourceManager/tests/NeutralResourcesLanguageAttributeTests.cs [hunk 2]
```diff
@@ -23,5 +25,25 @@ public static void ConstructorArgumentNull()
...
+        [InlineData(typeof(System.Collections.Concurrent.ConcurrentBag<string>))] // System.Collections.Concurrent
+        [InlineData(typeof(System.Console))] // System.Console
+        [InlineData(typeof(System.IO.Directory))] // System.IO.FileSystem
+        [InlineData(typeof(System.Linq.Enumerable))] // System.Linq
+        [InlineData(typeof(System.Net.Http.HttpClient))] // System.Net.Http
+        [InlineData(typeof(System.Text.RegularExpressions.Regex))] // System.Text.RegularExpressions
+        [InlineData(typeof(System.Threading.Barrier))] // System.Threading
+        public static void TestAttributeExistence(Type type)
+        {
+            CustomAttributeData attribute = type.Assembly.CustomAttributes.FirstOrDefault(a => a.AttributeType.Name == "NeutralResourcesLanguageAttribute");
```
Comment (ericstj) [inline src/libraries/System.Resources.ResourceManager/tests/NeutralResourcesLanguageAttributeTests.cs:43]: Consider making this test case call `ResourceManager.GetNeutralResourcesLanguage` as that was the API reported as being broken in the original issue. https://github.com/dotnet/runtime/issues/13349#issuecomment-536176848

File: eng/versioning.targets [hunk 3]
```diff
@@ -57,6 +57,10 @@
         <_Parameter1>$(AssemblyName)</_Parameter1>
       </AssemblyAttribute>
 
+      <AssemblyAttribute Include="System.Resources.NeutralResourcesLanguage" Condition="'$(IsTestProject)' != 'true' and '$(IsTestSupportProject)' != 'true'">
```
Comment (ericstj) [inline eng/versioning.targets:60]: Would it make more sense to base the condition on whether or not the assembly contains resources?  I see lots of build failures due to including this in reference assemblies which weren't referencing ResourceManager.

Perhaps `'@(EmbeddedResource)' != '' and '$(IsSourceProject)' == 'true'`?

File: src/libraries/System.Resources.ResourceManager/tests/NeutralResourcesLanguageAttributeTests.cs (hunk 2 above)
Comment (tarekgh) [reply src/libraries/System.Resources.ResourceManager/tests/NeutralResourcesLanguageAttributeTests.cs:43]: I'll try that but wanted to tell GetNeutralResourcesLanguage is protected so I'll still need to use the reflection anyway.

---
In reply to: [453763004](https://github.com/dotnet/runtime/pull/39135#discussion_r453763004) [](ancestors = 453763004)

Comment (ericstj) [reply src/libraries/System.Resources.ResourceManager/tests/NeutralResourcesLanguageAttributeTests.cs:43]: protected static, that's interesting.  You can call it like this:
```c#
public class MyResourceManager : System.Resources.ResourceManager
{
  public static CultureInfo GetNeutralResourcesLanguage(Assembly a)
  {
    return ResourceManager.GetNeutralResourcesLanguage(a);
  }
}
```

Comment (tarekgh) [reply src/libraries/System.Resources.ResourceManager/tests/NeutralResourcesLanguageAttributeTests.cs:43]: yes that will work too. I'll try that.

---
In reply to: [453795934](https://github.com/dotnet/runtime/pull/39135#discussion_r453795934) [](ancestors = 453795934)

File: src/coreclr/src/System.Private.CoreLib/src/System/Internal.cs (hunk 1 above)
Comment (tarekgh) [reply src/coreclr/src/System.Private.CoreLib/src/System/Internal.cs:13]: done!

File: src/libraries/System.Resources.ResourceManager/tests/NeutralResourcesLanguageAttributeTests.cs (hunk 2 above)
Comment (tarekgh) [reply src/libraries/System.Resources.ResourceManager/tests/NeutralResourcesLanguageAttributeTests.cs]: Done!

File: eng/versioning.targets (hunk 3 above)
Comment (tarekgh) [reply eng/versioning.targets]: Done!

**Review by safern — APPROVED** [review]

**Comment by tarekgh** [comment]
The CI leg `runtime (Libraries Test Run release mono OSX x64 Debug)` is unrelated and tracked by the issue https://github.com/dotnet/runtime/issues/38943

**Pull request merged**